pallets/loans/src/functions.rs

```diff
@@ -124,8 +124,12 @@ impl<T: Config> Pallet<T> {
 		Loan::<T>::try_mutate(pool_id, loan_id, |loan| -> DispatchResult {
 			let loan = loan.as_mut().ok_or(Error::<T>::MissingLoan)?;
 
-			// ensure loan is created
-			ensure!(loan.status == LoanStatus::Created, Error::<T>::LoanIsActive);
+			// ensure loan is created or priced but not yet borrowed against
+			ensure!(
+				loan.status == LoanStatus::Created
+					|| loan.status == LoanStatus::Active && loan.total_borrowed == Zero::zero(),
+				Error::<T>::LoanIsActive
+			);
```

Comment:
Just thought about this a little. What happens when I have repaid everything (i.e. `total_borrowed = 0`, `status = Active` and I then call this function? 

Could this result in weird behavior?

Reply:
Hm, maybe we need to rename `total_borrowed`. But this is an aggregated value, so it's not decreased on repayment (on purpose, because with a bullet loan, you shouldn't be able to borrow again after you've repaid the full value of the loan).

Reply:
Then I am fine with this!